lisp/ess-custom.el

```diff
@@ -2647,20 +2647,28 @@ This variable has no effect. Customize
 
 (defvar ess-R-keywords
   '("if" "else" "repeat" "while" "function" "for" "in" "next" "break")
-  "Reserved words in the R language.")
+  "Reserved words in the R language.
+If you modify this variable, call `ess-r-generate-font-lock-regexps'
+for the changes to take effect.")
 
 (defvar ess-R-control-flow-keywords
   '("switch" "function" "return" "on.exit" "stop"
     "tryCatch" "withRestarts" "invokeRestart"
     "recover" "browser")
   "Keywords that impact control flow.
 These keywords either cause a control flow jump or establish a
-jump target.")
+jump target.
+
+If you modify this variable, call `ess-r-generate-font-lock-regexps'
+for the changes to take effect.")
 
 (defvar ess-R-signal-keywords
   '("message" "warning" "signalCondition" "withCallingHandlers")
   "Keywords for condition signalling.
-These keywords might cause a control flow jump but do not necessarily.")
+These keywords might cause a control flow jump but do not necessarily.
+
```

Comment:
A side note. let's try to avoid fragmentation of doc strings. For long docs it might be warranted but not here. Blanks eat vertical space, interfere with structural navigation and fragment the code. Plus it's not the best practice to create one-sentence-paragraphs. Lisp is beautiful because it's compact, and it's a pity if rarely read docs would spoil that.

Reply:
Also I think the general solution for navigability of elisp code is to have the docstring folded by default.

Reply:
I don't know why people think that having a sentence in a separate paragraph improves readability. In any case, this one is not a defcustom, but an internal var. We should strive to keep internal vars and functions documented to the minimum. As a developer you read the code 100 times and the docs once.

> Also I think the general solution for navigability of elisp code is to have the docstring folded by default.

True. Do you happen to have a good setup for this?